### PR TITLE
Fix the test for population.CalculateRecruitment()

### PR DESCRIPTION
**Pull request by Bai-Li-NOAA**
# What is the feature?

This PR addresses issue #494 and fixes the test for population.CalculateRecruitment().

# How have you implemented the solution?

- update the C++ test [here](https://github.com/NOAA-FIMS/FIMS/blob/3dbecd32218697b992877f2395b6ada8d3587588/tests/gtest/test_population_Recruitment.cpp)
- register the test to [CMakeLists.txt](https://github.com/NOAA-FIMS/FIMS/blob/3dbecd32218697b992877f2395b6ada8d3587588/tests/gtest/CMakeLists.txt#L246C1-L256C45) to execute the test when running GoogleTest

# Does the PR impact any other area of the project?

No.

# How to test this change

The C++ test can be found [here](https://github.com/NOAA-FIMS/FIMS/blob/3dbecd32218697b992877f2395b6ada8d3587588/tests/gtest/test_population_Recruitment.cpp).
[All GHA checks](https://github.com/NOAA-FIMS/FIMS/actions?query=branch%3A494-questionhelp-why-is-populationspawning_biomass-indexed-by-age-in-test) have passed. 

# Developer pre-PR checklist

- [x] I relied on GitHub actions to :test_tube: things for me while I sat on the :couch_and_lamp:.

### CODE REVIEW

File: inst/include/population_dynamics/population/population.hpp [hunk 1]
```diff
@@ -377,7 +377,7 @@ struct Population : public fims_model_object::FIMSObject<Type> {
     Type phi0 = CalculateSBPR0();
     FIMS_LOG << "recruitment 2" << std::endl;
     FIMS_LOG << "phi0 = " << phi0 << std::endl;
-    FIMS_LOG << "spawning biomass = " << this->spawning_biomass[year]
+    FIMS_LOG << "spawning biomass = " << this->spawning_biomass[year - 1]
```
Comment (iantaylor-NOAA) [inline inst/include/population_dynamics/population/population.hpp:380]: This change seems to match what's noted in https://noaa-fims.github.io/collaborative_workflow/m1-model-specification.html#beverton-holt-recruitment-function, but I might change it to 
"spawning_biomass[year - 1] = " to make it extra clear what's being logged.

File: tests/gtest/test_population_Recruitment.cpp [hunk 2]
```diff
@@ -8,70 +8,49 @@ namespace
...
+        population.CalculateSpawningBiomass(sb_i_age_year, sb_year, sb_age);
+
+        // calculating phi0
+        double phi0 = population.CalculateSBPR0();
+    
+        // calculating recruitment for year 5
+        int r_year = 5;
+        int r_age = 0; // just calculating for age 0 
+        int r_i_age_year = r_year * population.nages + r_age;
+        int r_i_agem1_yearm1 = (r_year - 1) * population.nages + r_age - 1;
```
Comment (iantaylor-NOAA) [inline tests/gtest/test_population_Recruitment.cpp:34]: Is this line needed? I don't see `r_i_agem1_yearm1` being used anywhere else in the file.

File: tests/gtest/test_population_Recruitment.cpp [hunk 3]
```diff
@@ -8,70 +8,49 @@ namespace
 
     TEST_F(PopulationPrepareTestFixture, CalculateRecruitment_works)
     {
-        // calculating recruitment for year 4
-        int year = 4;
-        int age = 0; // just calculating for age 0 
+        // calculating spawning biomass for year 4
+        int sb_year = 4;
```
Comment (iantaylor-NOAA) [inline tests/gtest/test_population_Recruitment.cpp:12]: Separating `sb_year = 4`  from `r_year = 5` makes sense to keep things clear.

**Review by iantaylor-NOAA — APPROVED** [review]
Changes look good. Thanks @Bai-Li-NOAA for sorting all this so quickly.

**Comment by iantaylor-NOAA** [comment]
Note: based on conversation during yesterday's meeting, now that I've completed the review, I've unassigned myself and assigned @Bai-Li-NOAA so she can choose how/whether to address the minor comments in the unresolved conversations. If I didn't have any comments, I would have just merged since I was the assignee. Once @Bai-Li-NOAA has resolved the conversations, she could either merge the PR or re-assign me if further input is needed. 

Please let me know if that workflow doesn't match how we're thinking of using PR assignments.

File: inst/include/population_dynamics/population/population.hpp (hunk 1 above)
Comment (Bai-Li-NOAA) [reply inst/include/population_dynamics/population/population.hpp]: Agree. The text has been changed to "spawning_biomass[year - 1] = ".

File: tests/gtest/test_population_Recruitment.cpp (hunk 2 above)
Comment (Bai-Li-NOAA) [reply tests/gtest/test_population_Recruitment.cpp]: No, `r_i_agem1_yearm1` is not needed. I have removed this line of code. Thanks for catching it.

**Comment by Bai-Li-NOAA** [comment]
Thank you, @iantaylor-NOAA, for your review. I've incorporated your suggestions and updated the code. Additionally, I've resolved the merge conflicts. The workflow makes sense to me and I have merged the changes into the main branch.

**Pull request merged**